python/vsi/tools/patch_wheel.py

```diff
@@ -1,77 +1,112 @@
+  parser.add_argument('--name', help="The package name")
   group = parser.add_mutually_exclusive_group()
   group.add_argument('--version', help="Version number to change to")
-  group.add_argument('--add-local', help="Add to the local version segment without having to specify the full version")
+  group.add_argument('--add-local',
+                     help="Add to the local version segment without having to "
+                          "specify the full version")
   return parser
 
 
-def main(filename, version=None):
-  filename = pathlib.Path(filename)
+def main(filename, name=None, version=None):
+  filename = Path(filename)
 
-  # Split to {normalize name}-{version}-{remainder}
-  normalized_name, file_version, remainder = filename.stem.split('-', 2)
+  parsed_name = WHEEL_INFO_RE.match(filename.name)
+  original_name, original_version = parsed_name.group('name', 'ver')
 
+  # These values will be replaced
   patch = {}
-  # if name is not None:
-  #   # https://peps.python.org/pep-0491/#escaping-and-unicode
-  #   normalized_name = re.sub(r"[^\w\d.]+", "_", name).lower()
-  #   patch[b'Name'] = b'Name: ' + name.encode()
+  if name is not None:
+    patch['Name'] = f'Name: {name}\n'
+  else:
+    name = original_name
   if version is not None:
-    patch[b'Version'] = b'Version: ' + version.encode()
+    patch['Version'] = f'Version: {version}\n'
   else:
-    version = file_version
-
-  new_filename = filename.parent / ('-'.join([normalized_name,
-                                              version,
-                                              remainder]) + '.whl')
-
-  with tempfile.NamedTemporaryFile(suffix='.whl',
-                                   prefix='.' + filename.stem + '.',
-                                   dir=filename.parent,
-                                   delete=False) as temp_file:
-    os.chmod(temp_file.name, 0o644) # Python hardcodes this to 600 :(
-    with zipfile.ZipFile(filename, 'r') as zin:
-      with zipfile.ZipFile(temp_file, 'w') as zout:
-        zout.comment = zin.comment # preserve the comment
-        for item in zin.infolist():
-          # Patch METADATA to match filename
-          if item.filename.endswith('.dist-info/METADATA'):
-            meta_lines = (patch[key]
-                          if (key := line.split(b':',1)[0]) in patch.keys()
-                          else line
-                          for line in zin.read(item.filename).split(b'\n'))
-            zout.writestr(item, b'\n'.join(meta_lines))
-          else:
-            zout.writestr(item, zin.read(item.filename))
-  os.rename(temp_file.name, new_filename)
+    version = original_version
+
+  # https://peps.python.org/pep-0491/#escaping-and-unicode
+  normalized_name = re.sub(r"[^\w\d.]+", "_", name).lower()
+  normalized_version = re.sub(r"[^\w\d.+]+", "_", version).lower()
+
+  if name == original_name and version == original_version:
```

Comment:
move this above `normalized_name`

Reply:
so basically, either `name` or version must be passed in? why not just create a pre-condition check

Reply:
> move this above `normalized_name`

Why?

> so basically, either `name` or version must be passed in? why not just create a pre-condition check

Well, you should be annoying and specify it and specify the same thing. This _is_ the pre-condition?